File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.py
formula=str(input())
temp = formula.split('-')
# -를 기준 -> 파씽
result = 0
# -로 시작하는 경우

if formula[0] == '-':
    temp.pop(0)
    k=sum(map(int,temp[0].split('+')))
    result -= k
else:
    k=sum(map(int,temp[0].split('+')))
    result += k

for i in temp[1:]:
    k=sum(map(int,i.split('+')))
    result-=k
print(result)

File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.cc
#include <iostream>
#include <queue>
#include <tuple>
#include <string>
#include <algorithm>;
using namespace std;


vector<int> num;
vector<char> oper;
void init(string input) {
	string temp = "";
	for (char ch : input) {
		if (isdigit(ch)) {
			temp += ch;
		}
		else {
			oper.push_back(ch);
			num.push_back(stoi(temp));
			temp = "";
		}
	}
	num.push_back(stoi(temp));
}
int main() {
	string input; cin >> input;
	init(input);
	// - 뒤에 또 - 나오기 전까지 쭉 괄호로 묶기
	
	
	int result = num[0];

	bool minusmode = false;
	for (int i = 0; i < oper.size();i++) {
		if (oper[i] == '-') {
			minusmode = true;
		}

		if (minusmode) {
			result -= num[i + 1];
		}
		else {
			result += num[i + 1];
		}
	}
	cout << result;
}